### ISSUE TITLE
docker pull failed on a large task array in Google cloud with Cromwell 85

### ISSUE THREAD

**Issue by freeseek** [broadinstitute/cromwell#7132]
While running a WDL on Google cloud with a very large task array I received the weirdest error:
```
"failures": [
          {
            "causedBy": [],
            "message": "Task xxx.xxx:2292:1 failed. The job was stopped before the command finished. PAPI error code 7. Execution failed: generic::permission_denied: pulling image: docker pull: running [\"docker\" \"pull\" \"us.gcr.io/mccarroll-mocha/bcftools@sha256:ef8a606bfa1c5cef455dc2b97812f0a7f16d89e2962eee7654457a371462194b\"]: exit status 1 (standard error: \"error pulling image configuration: download failed after attempts=1: error parsing HTTP 403 response body: invalid character '<' looking for beginning of value: \\\"<?xml version='1.0' encoding='UTF-8'?><Error><Code>AccessDenied</Code><Message>Access denied.</Message><Details>We're sorry, but this service is not available in your location</Details></Error>\\\"\\n\")"
          }
        ],
```
And the shard-2292 directory only contained the following files (no `stderr` file):
```
gcs_delocalization.sh
gcs_localization.sh
gcs_transfer.sh
script
```
The task array is very large (0-2437) but only three different tasks had this error (1124, 1510, 2292). It seems as if Cromwell tried to pull the docker (`us.gcr.io/mccarroll-mocha/bcftools:1.16-20221221` in this case) from GCR but received a 403 response for some reasons and did not know how to interpret it. When rerunning the pipeline with CallCaching, it continued without issues.

My suggestion is that this is a rare failure mode that might only occur for large task arrays and Cromwell needs to interpret the 403 error and retry to pull the docker after a few seconds rather than failing the whole pipeline.

**Comment by freeseek** [comment]
I see, so is the VM tasked through PAPI to pull the docker and then run `gcs_localization.sh` and `script`? Is it the internal PAPI code that failed to properly handle the 403 error when pulling the docker?

**Comment by aednichols** [comment]
Yeah, that sounds right to me. I'm not sure how PAPI could improve though, seeing as 4xx errors are generally not retryable.

We do have logic in Cromwell to detect certain common pull failures [1], but I'm not sure this one [0] is common enough to merit inclusion as a special case.

[0] `We're sorry, but this service is not available in your location`
[1] https://github.com/broadinstitute/cromwell/blob/d142bd4a9b890605a2e61ee6469f01a442dfb74e/supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiAsyncBackendJobExecutionActor.scala#L784-L813

**Comment by aofarrel** [comment]
> Yeah, that sounds right to me. I'm not sure how PAPI could improve though, seeing as 4xx errors are generally not retryable.

For the "docker pull failed during a very large scatter" case, retries are pretty much always successful (except when the image actually doesn't exist of course). Would it be reasonable for this to be handled, especially since this isn't a rare failure when scattering more than 300x?
